From 4b7a99f5f7dcd735610b8085fbbc365bb9109a1f Mon Sep 17 00:00:00 2001
From: Zhou Furong <furong.zhou@intel.com>
Date: Wed, 6 Dec 2017 09:26:03 +0800
Subject: [PATCH 0863/2302] net:bcmdhd: fix depencency of CONFIG_PM and
 CONFIG_PM_SLEEP

dhd_mmc_suspend defined only when CONFIG_PM_SLEEP enabled

Change-Id: I5d18c4911fd78c282b43ce9b11e8cc3da11795d9
---
 drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
index db41c59..19fa8b5 100644
--- a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
+++ b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc_linux.c
@@ -231,18 +231,25 @@ static int bcmsdh_sdmmc_suspend(struct device *pdev)
 	if (func->num != 2)
 		return 0;
 
+#ifdef CONFIG_PM_SLEEP
 	dhd_mmc_suspend = TRUE;
+#endif
+
 	sdioh = sdio_get_drvdata(func);
 	err = bcmsdh_suspend(sdioh->bcmsdh);
 	if (err) {
+#ifdef CONFIG_PM_SLEEP
 		dhd_mmc_suspend = FALSE;
+#endif
 		return err;
 	}
 
 	sdio_flags = sdio_get_host_pm_caps(func);
 	if (!(sdio_flags & MMC_PM_KEEP_POWER)) {
 		sd_err(("%s: can't keep power while host is suspended\n", __FUNCTION__));
+#ifdef CONFIG_PM_SLEEP
 		dhd_mmc_suspend = FALSE;
+#endif
 		return  -EINVAL;
 	}
 
@@ -250,7 +257,9 @@ static int bcmsdh_sdmmc_suspend(struct device *pdev)
 	err = sdio_set_host_pm_flags(func, MMC_PM_KEEP_POWER);
 	if (err) {
 		sd_err(("%s: error while trying to keep power\n", __FUNCTION__));
+#ifdef CONFIG_PM_SLEEP
 		dhd_mmc_suspend = FALSE;
+#endif
 		return err;
 	}
 	smp_mb();
@@ -268,7 +277,9 @@ static int bcmsdh_sdmmc_resume(struct device *pdev)
 		return 0;
 
 	sdioh = sdio_get_drvdata(func);
+#ifdef CONFIG_PM_SLEEP
 	dhd_mmc_suspend = FALSE;
+#endif
 
 	smp_mb();
 	return 0;
-- 
2.7.4

